client/src/PaymentPage/Payment.jsx

```diff
@@ -115,156 +104,222 @@ function Payment() {
+    outboundStops.push(selectedOutbound.departure.id);
+    for (const stop of context.SelectedTrain.stops) {
+      outboundStops.push(stop.id);
+    }
+    outboundStops.push(selectedOutbound.arrival.id);
+
+    const outboundBooking = await fetch(API_URL + "/booking/reservation", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        "Access-Control-Allow-Origin": "*",
+      },
+      body: JSON.stringify({
+        customer: {
+          firstName: customerFirst,
+          lastName: customerLast,
+          email: customerEmail,
+          phoneNumber: customerPhone,
+        },
+        trainStops: outboundStops,
+        seats: tickets,
+      }),
+    });
+
+    let returnBooking;
+    if (context?.searchData?.returnTrip) {
+      const selectedReturn = context.dbData?.ReturnTrips?.filter(
+        (trip) => trip.train.id === context.SelectedTrain.trainID
+      )[0];
+
+      const returnStops = [];
+      returnStops.push(selectedReturn.departure.id);
+      for (const stop of context.SelectedReturnTrain.stops) {
+        returnStops.push(stop.id);
+      }
+      returnStops.push(selectedReturn.arrival.id);
+
+      returnBooking = await fetch(API_URL + "/booking/reservation", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          "Access-Control-Allow-Origin": "*",
+        },
+        body: JSON.stringify({
+          customer: {
+            firstName: customerFirst,
+            lastName: customerLast,
+            email: customerEmail,
+            phoneNumber: customerPhone,
+          },
+          trainStops: returnStops,
+          seats: tickets,
+        }),
+      });
+    }
+
+    const parsedOutbound = await outboundBooking.json();
+    const parsedReturn = await returnBooking.json();
+    fetch(API_URL + "/booking/create-checkout-session", {
+      method: "POST",
       headers: {
-        'Content-Type': 'application/json',
-        'Access-Control-Allow-Origin': '*',
+        "Content-Type": "application/json",
+        "Access-Control-Allow-Origin": "*",
       },
       body: JSON.stringify({
         items: items,
+        outboundBookingId: parsedOutbound.data.booking.id,
+        returnBookingId: parsedReturn?.data?.booking?.id,
       }),
     })
-      .then((res) => {
+      .then(async (res) => {
         if (res.ok) {
           return res.json();
         } else {
-          return res.json().then((json) => Promise.reject(json));
+          const json = await res.json();
+          return await Promise.reject(json);
         }
       })
       .then((data) => {
         window.location = data.data.url;
       })
       .catch((e) => {
-        console.log(e.error);
+        console.log("Fetch checkout-session error:", e.error);
       });
   }
 
   function returnPage() {
-    if (context.SelectedTrain.class === 'FirstClass') {
-      nav('/results');
+    if (context.SelectedTrain.class === "FirstClass") {
```

Comment:
You forgot to change it to "First Class" here, it goes back to additionalpage even if i've selected firstclass tickets.

Reply:
Good catch, issue has been fixed.